src/main/java/org/dice/factcheckbenchmark/benchmark/rdf/TripleExtractor.java

```diff
@@ -0,0 +1,190 @@
+import java.io.FileNotFoundException;
+
+/**
+ * @author DANISH AHMED on 3/22/2018
+ * Extraction idea taken from: https://github.com/SmartDataAnalytics/DeFacto
+ * and then modified/optimized according to our need
+ */
+public class TripleExtractor {
+
+    private Model model;     // provides model after file has been read
+    private RDFResource subject;
+    private Property predicate;
+    private RDFResource object;
+
+    private String subjectUri;
+    private String predicateUri;
+    private String objectUri;
+
+    private Model simplifiedModel;
+    private String simplifiedData;
+
+    public TripleExtractor(String fileName) throws FileNotFoundException {
+        Model model = ModelFactory.createDefaultModel();
+        model.read(new FileInputStream(fileName), null, "TTL");
+
+        setModel(model);
+        parseStatements();
+        setUris();
+
+        setSimplifiedData();
+    }
+
+    private void setUris() {
+        subjectUri = getResourceUri(subject);
+        objectUri = getResourceUri(object);
+        predicateUri = predicate.getURI();
+    }
+
+    private void setModel(Model model) {
+        this.model = model;
+    }
+
+    private void setSimplifiedData() {
+
+        simplifiedData = String.format("<http://swc2017.aksw.org/task/dataset/s> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/1999/02/22-rdf-syntax-ns#Statement> .\n" +
+                "<http://swc2017.aksw.org/task/dataset/s> <http://www.w3.org/1999/02/22-rdf-syntax-ns#subject> <%s> .\n" +
+                "<http://swc2017.aksw.org/task/dataset/s> <http://www.w3.org/1999/02/22-rdf-syntax-ns#predicate> <%s> .\n" +
+                "<http://swc2017.aksw.org/task/dataset/s> <http://www.w3.org/1999/02/22-rdf-syntax-ns#object> "
+                , subjectUri
+                , predicateUri);
+
+        if (object.resource.isResource())
+            simplifiedData = simplifiedData + String.format("<%s> .\n", objectUri);
+        else if (object.resource.isLiteral())
+            simplifiedData = simplifiedData + String.format("\"%s\" .\n", objectUri);
+
+        String labels = String.format("<%s> <http://www.w3.org/2000/01/rdf-schema#label> \"%s\"@en .\n" +
+                "<%s> <http://www.w3.org/2000/01/rdf-schema#label> \"%s\"@en .\n"
+                , subjectUri, subject.label
+                , objectUri, object.label);
+        simplifiedData = simplifiedData + labels;
+    }
+
+    public Model getSimplifiedModel() {
+        return this.simplifiedModel;
+    }
+
+    public String getSimplifiedData() {
+        return this.simplifiedData;
+    }
+
+    private void setSimplifiedModel() {
+        simplifiedModel = ModelFactory.createDefaultModel();
+        simplifiedModel.read(new ByteArrayInputStream(simplifiedData.getBytes()), null, "TTL");
+    }
+
+    private void parseStatements() {
+        boolean startNode = false;
+
+        StmtIterator stmtIterator = this.model.listStatements();
```

Comment:
using `listSubjects()` to search for the resource which matches your regex would be easier... right?